sdk/src/main/java/com/uber/sdk/android/rides/RequestButton.java

```diff
@@ -92,22 +104,26 @@ protected void init(
 
         super.init(context, attributeSet, defStyleAttrs, defStyleRes);
 
-        setOnClickListener(new OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                if (mClientId == null) {
-                    throw new IllegalStateException("Client ID required to use RequestButton.");
-                }
+        setOnClickListener(this);
+    }
 
-                RequestDeeplink requestDeeplink = new RequestDeeplink.Builder()
-                        .setClientId(mClientId)
-                        .setRideParameters(mRideParameters)
-                        .setUserAgent(USER_AGENT_BUTTON)
-                        .build();
+    @Override
+    public void onClick(View v) {
```

Comment:
Instead of having this override onClick, create a RequestOnClick that is set as the internal callback